src/crypto/yespower-bitzeny.c

```diff
@@ -0,0 +1,50 @@
+/*-
+ * Copyright 2009 Colin Percival
+ * Copyright 2013-2018 Alexander Peslyak
+ * Copyright 2018 bitzenyPlus Core Developers
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ * This file was originally written by Colin Percival as part of the Tarsnap
+ * online backup system.
+ */
+
+#include "yespower/insecure_memzero.h"
+#include "yespower/sha256.c"
+#include "yespower/sha256.h"
+#include "yespower/yespower.h"
+#include "yespower/yespower-opt.c"
+
+static const yespower_params_t yespower_0_5_bitzeny = {YESPOWER_0_5, 2048, 8, "Client Key", 10};
+
+// WARNING DO NOT USE THIS IN yespower_hash YET. For later HF to yespower 1.0
+// TODO Find HF solution using unixtime
```

Comment:
cryply team uses `unixtime`. is this a good way for HF solution? we need the method from `yesPower 0.5` to `1.0`.

https://github.com/cryply/cryply-wallet/blob/d974128611963e664232c1d24a01f9b284af7421/src/hash/yespower/yespower.c#L39

```c
static const yespower_params_t v1 = {YESPOWER_0_5, 4096, 16, "Client Key", 10};

static const yespower_params_t v2 = {YESPOWER_0_9, 2048, 32, NULL, 0};

int yespower_hash(const char *input, char *output)
{
    uint32_t time = le32dec(&input[68]);
    if (time > 1530403200) {
        return yespower_tls(input, 80, &v2, (yespower_binary_t *) output);
    } else {
        return yespower_tls(input, 80, &v1, (yespower_binary_t *) output);
    }
}
```

Reply:
you are right. this code is not urgent for now. however i just wanted to hear solardiz's opinion.

Reply:
I think basing the hard-fork on block height is slightly safer, but once the network successfully gets significantly past this event it doesn't matter which method was used - there's supposed to be core code that disallows decreasing block timestamps.

Reply:
>I think basing the hard-fork on block height is safer

I think so.
We should consider again when the hard fork is finalized. 
It is better that not to change now.

Reply:
i agree. not now. we need more discussion about this. it's very important thing to our future. :thinking: